tests/SpecTests/ChangeStreamsSpecTest.php

```diff
@@ -18,6 +18,10 @@
  */
 class ChangeStreamsSpecTest extends FunctionalTestCase
 {
+    private static $incompleteTests = [
+        'change-streams-errors: Change Stream should error when _id is projected out' => 'PHPC-1419',
+    ];
```

Comment:
Was this test previously passing?

Reply:
It was passing, then it was failing after the update. Now after trying to reproduce it, it's no longer failing, so go figure. I'll unskip it and let travis decide what's going on.

FWIW, until [PHPC-1419](https://jira.mongodb.org/browse/PHPC-1419) is fixed we'll try to resume the change stream after `getMore` failed with the expected error, with the subsequent `aggregate` command failing as expected. This is fixed when we finish up sharing tests in #659.

Reply:
The test fails on sharded clusters, but for different reasons.

On a sharded cluster backed by standalone servers, `$changeStream` obviously isn't supported. This causes an exception when trying to run the watch operation from the spec test, but that exception does not contain the `NonResumableChangeStreamError` (because it's not a change stream error) label, causing the test to fail.
This is caused because the spec requires this test to run on both replicaset and sharded topologies. Unfortunately, the spec does not differentiate between the two types of sharded clusters (backed entirely by replicasets or not), so this is something we may want to address in the specification in the future. However, since the spec test also require testing change streams on a standalone server to ensure an exception is thrown, we can't just skip these tests on unsupported platforms. I may have to hardcode an exception for this one particular test.

On a sharded cluster backed by replica sets, we're seeing the behaviour described in https://github.com/mongodb/mongo-php-library/pull/659#issue-304180039. In a nutshell, the first `getMore` call does not return any results as mongos hasn't received returnable results from all shards yet. A subsequent `getMore` would return a document, causing the `getMore` call to fail with the expected error. Even then, the test still fails because of [PHPC-1419](https://jira.mongodb.org/browse/PHPC-1419): the `NonResumableChangeStreamError` is not present in the exception.

With that in mind, I'd skip the test for now and properly fix it in #659 where we take care of other change stream tests as well.

FWIW, debugging this issue made me realise that PHPC-1419 only appears on sharded clusters. I'll investigate that further separately to figure out the cause of the bug.